src/env.rs

```diff
@@ -36,6 +37,17 @@ pub fn is_uhyve() -> bool {
 	matches!(boot_info().platform_info, PlatformInfo::Uhyve { .. })
 }
 
+pub fn fdt() -> Option<Fdt<'static>> {
+	kernel::boot_info().hardware_info.device_tree.map(|fdt| {
+		let ptr = ptr::with_exposed_provenance(fdt.get().try_into().unwrap());
+		unsafe { Fdt::from_ptr(ptr).unwrap() }
+	})
+}
+
+pub fn fdt_args() -> Option<&'static str> {
```

Comment:
I like brief doc comments on functions, but this is a personal preference.

Shouldn't these functions be `pub (crate)`?

Reply:
> Shouldn't these functions be `pub (crate)`?

I prefer not setting the export scope when the whole module is private.